src/vdaf/dummy.rs

```diff
@@ -0,0 +1,327 @@
+            .finish()
+    }
+}
+
+impl Vdaf {
+    /// The length of the verify key parameter for fake VDAF instantiations.
+    pub const VERIFY_KEY_LEN: usize = 0;
+
+    /// Construct a new instance of the dummy VDAF.
+    pub fn new(rounds: u32) -> Self {
+        Self {
+            rounds,
+            prep_init_fn: Arc::new(|_| -> Result<(), VdafError> { Ok(()) }),
+            prep_step_fn: Arc::new(
+                |state| -> Result<PrepareTransition<Self, 0, 16>, VdafError> {
+                    let new_round = state.current_round + 1;
+                    if new_round == state.max_rounds {
+                        Ok(PrepareTransition::Finish(OutputShare(state.input_share)))
+                    } else {
+                        Ok(PrepareTransition::Continue(
+                            PrepareState {
+                                current_round: new_round,
+                                ..*state
+                            },
+                            (),
+                        ))
+                    }
+                },
+            ),
+        }
+    }
+
+    /// Provide an alternate implementation of
+    /// [`vdaf::Aggregator::prepare_init`].
+    pub fn with_prep_init_fn<F: Fn(&AggregationParam) -> Result<(), VdafError>>(
+        mut self,
+        f: F,
+    ) -> Self
+    where
+        F: 'static + Send + Sync,
+    {
+        self.prep_init_fn = Arc::new(f);
+        self
+    }
+
+    /// Provide an alternate implementation of
+    /// [`vdaf::Aggregator::prepare_step`].
+    pub fn with_prep_step_fn<
+        F: Fn(&PrepareState) -> Result<PrepareTransition<Self, 0, 16>, VdafError>,
+    >(
+        mut self,
+        f: F,
+    ) -> Self
+    where
+        F: 'static + Send + Sync,
+    {
+        self.prep_step_fn = Arc::new(f);
+        self
+    }
+}
+
+impl Default for Vdaf {
+    fn default() -> Self {
+        Self::new(1)
+    }
+}
+
+impl vdaf::Vdaf for Vdaf {
+    const ID: u32 = 0xFFFF0000;
+
+    type Measurement = u8;
+    type AggregateResult = u8;
+    type AggregationParam = AggregationParam;
+    type PublicShare = ();
+    type InputShare = InputShare;
+    type OutputShare = OutputShare;
+    type AggregateShare = AggregateShare;
+
+    fn num_aggregators(&self) -> usize {
+        2
+    }
+}
+
+impl vdaf::Aggregator<0, 16> for Vdaf {
+    type PrepareState = PrepareState;
+    type PrepareShare = ();
+    type PrepareMessage = ();
+
+    fn prepare_init(
+        &self,
+        _verify_key: &[u8; 0],
+        _: usize,
+        aggregation_param: &Self::AggregationParam,
+        _nonce: &[u8; 16],
+        _: &Self::PublicShare,
+        input_share: &Self::InputShare,
+    ) -> Result<(Self::PrepareState, Self::PrepareShare), VdafError> {
+        (self.prep_init_fn)(aggregation_param)?;
+        Ok((
+            PrepareState {
+                input_share: input_share.0,
+                current_round: 0,
+                max_rounds: self.rounds,
```

Comment:
Q: why do we need to track `max_rounds` in `prepare_state` -- couldn't we just have the default `prep_step_fn` use `rounds`? (i.e. replace `state.max_rounds` on L53 with `rounds`)